lib/interceptors/csrf-token.ts

```diff
@@ -0,0 +1,29 @@
+import { generateUrl } from '@nextcloud/router'
+
+const RETRY_KEY = Symbol('csrf-retry')
+
+export const onError = axios => async (error) => {
+	const { config, response, request: { responseURL } } = error
+	const { status } = response
+
+	if (status === 412
+		&& response?.data?.message === 'CSRF check failed'
+		&& config[RETRY_KEY] === undefined) {
+		console.warn(`Request to ${responseURL} failed because of a CSRF mismatch. Fetching a new token`)
+
+		const { data: { token } } = await axios.get(generateUrl('/csrftoken'))
```

Comment:
**Question:** Does this endpoint exist in all versions of nextcloud?

Reply:
Yes, the controller was created in 2018 so the support is reasonable.